core/src/main/java/tc/oc/pgm/map/MapLibraryImpl.java

```diff
@@ -226,16 +223,22 @@ private MapContext loadMap(MapSource source, @Nullable String mapId) throws MapE
     }
 
     MapInfo info = context.getInfo();
-    maps.merge(
-        info.getId(), info, (m1, m2) -> m2.getVersion().isOlderThan(m1.getVersion()) ? m1 : m2);
+    // Only if from a supported version, add it to our library
+    if (info.isServerSupported()) {
+      maps.merge(
```

Comment:
It will simply skip maps that are not for the current server version. It is intended as you may maintain a repo that contains both 1.8 and 1.21 maps, and simply when running in 1.8 the 1.21 maps are skipped and all is good

I'm not sure about logging as it'd likely lead to log-spam that is not really serving any use, as in, whoever set those maps knows what they're doing